src/lib/detection_algorithms/cntk/exporter.js

```diff
@@ -0,0 +1,158 @@
+const async = require('async');
+const fs = require('fs');
+const path = require('path');
+
+// The Exporter interface - provides a mean to export the tagged frames
+// data in the expected data format of the detection algorithm
+// Constructor parameters:
+//  exportDirPath - path to the directory where the exported file will be placed
+//  classes - list of classes supported by the tagged data
+//  posFramesCount - number of positive tagged frames
+//  frameWidth - The width (in pixels) of the image frame
+//  frameHeight - The height (in pixels) of the image frame
+//  testSplit - the percent of tragged frames to reserve for test set defaults to 20%
+function Exporter(exportDirPath, classes, posFramesCount, frameWidth, frameHeight, testSplit) {
+    var self = this;
+    self.exportDirPath = exportDirPath;
+    self.posDirPath  = path.join(self.exportDirPath, 'positive');
+    self.negDirPath  = path.join(self.exportDirPath, 'negative');
+    self.testDirPath = path.join(self.exportDirPath, 'testImages');
+    self.classes = classes;
+    self.posFramesCount = posFramesCount;
+    self.frameWidth = frameWidth;
+    self.frameHeight = frameHeight;
+    self.testSplit = testSplit || 0.2;
+    self.posFrameIndex = null;
+    self.testFrameIndecies = null;
+
+    // Prepare everything for exporting (e.g. create metadata files,
+    // directories, ..)    
+    // Returns: A Promise object that resolves when the operation completes
+    this.init = function init() {
+
+         return new Promise(function(resolve, reject) {
+            async.waterfall([
+                generateTestIndecies.bind(null,self.testSplit),
+                function updateIndecies(testIndecies, cb) {
+                    self.posFrameIndex = 0;
+                    self.testFrameIndecies = testIndecies;
+                    cb();
+                },
+                ensureDirExists.bind(null, self.exportDirPath),
```

Comment:
You can do here some very ugly javascript magic, with Map 
(e.g. [self.exportDirPath, self.posDirPath..].map(=>(item) {return ensureDirExists.bind(null, item)} )
or something similar. Very not critical though.